kratos/solving_strategies/schemes/residual_based_adjoint_bossak_scheme.h

```diff
@@ -501,77 +846,139 @@ class ResidualBasedAdjointBossakScheme : public Scheme<TSparseSpace, TDenseSpace
                                       std::vector<const VariableData*>& rVec) {
                 return rExtensions.GetSecondDerivativesVariables(rVec);
             });
+        auto auxiliary_vars = GatherVariables(
+            rModelPart.Elements(), [](const AdjointExtensions& rExtensions,
+                                      std::vector<const VariableData*>& rVec) {
+                return rExtensions.GetAuxiliaryVariables(rVec);
+            });
+
         SetToZero_AdjointVars(lambda2_vars, rModelPart.Nodes());
         SetToZero_AdjointVars(lambda3_vars, rModelPart.Nodes());
 
-        const int number_of_elements = rModelPart.NumberOfElements();
-        const ProcessInfo& r_process_info = rModelPart.GetProcessInfo();
+        const auto& r_process_info = rModelPart.GetProcessInfo();
+        UpdateEntityTimeSchemeContributions(rModelPart.Elements(), r_process_info);
+        UpdateEntityTimeSchemeContributions(rModelPart.Conditions(), r_process_info);
+
+        // Finalize global assembly
+        Assemble_AdjointVars(lambda2_vars, rModelPart.GetCommunicator());
+        Assemble_AdjointVars(lambda3_vars, rModelPart.GetCommunicator());
+
+        for (unsigned int i_var = 0; i_var < lambda2_vars.size(); ++i_var) {
+            const auto& r_lambda2_variable_name = lambda2_vars[i_var]->Name();
+            const auto& r_lambda3_variable_name = lambda3_vars[i_var]->Name();
+            const auto& r_auxiliary_variable_name = auxiliary_vars[i_var]->Name();
+
+            if (KratosComponents<Variable<array_1d<double, 3>>>::Has(r_lambda2_variable_name)) {
```

Comment:
is this check really necessary as you operate with the variables already in the lines before?

Reply:
this is resolved after discussing with @MFusseder  :)